web/src/components/single/Metadata.tsx

```diff
@@ -0,0 +1,229 @@
+import React, { useEffect, useState } from "react";
+import { API } from "aws-amplify";
+import { Box, Button, Header, Table, Input } from "@cloudscape-design/components";
+
+class MetadataApi {
+    version!: string;
+    metadata!: Metadata;
+}
+class Metadata {
+    [key: string]: string;
+}
+
+class TableRow {
+    idx!: number;
+    name: string | null | undefined;
```

Comment:
Why are these both null || undefined?

Reply:
The name/description fields are empty when new rows are added by the user.